internal/store/db/redis/store.go

```diff
@@ -201,11 +201,18 @@ func (c Client) Disconnect() error {
 
 // NewClient provides a factory for building a StoreClient
 func NewClient(config db.DatabaseInfo) (interfaces.StoreClient, error) {
+	var retErr error
 	once.Do(func() {
 		connectionString := fmt.Sprintf("%s:%d", config.Host, config.Port)
+		connectTimeout, err := time.ParseDuration(config.Timeout)
+		if err != nil {
+			retErr = fmt.Errorf("config.Timeout failed to parse: %v", err)
+			return
+		}
+
 		opts := []redis.DialOption{
 			redis.DialPassword(config.Password),
-			redis.DialConnectTimeout(time.Duration(config.Timeout) * time.Millisecond),
+			redis.DialConnectTimeout(connectTimeout),
```

Comment:
Does this need to be `connectTimeout/time.Millisecond` or does it take `Duration`

Reply:
It takes `time.Duration`. 
```go
func DialConnectTimeout(d time.Duration) DialOption
```